src/map/atcommand.c

```diff
@@ -8544,6 +8544,9 @@ ACMD(set)
 			data->type = C_STR;
 			data->u.mutstr = aStrdup(str);
 		}
+	} else if (script->str_data[script->search_str(reg)].type == C_PARAM) {
```

Comment:
script->search_str may reeturn -1 if string not found, and here it will be buffer underflow.
probably better assign result of script->search_str into variable inside elseif and after check type